torchtitan/checkpoint.py

```diff
@@ -220,44 +176,29 @@ def __init__(
 
             TODO: This is currently unsolved and needs a fix.
         """
-        assert len(model_parts) == len(
+        if job_config.optimizer.early_step_in_backward:
```

Comment:
We make the optimizers.optimizers to be plain for both in backward or not in.
Thus `len(model_parts)` does not work for in backward case.
What do you think of the assert check here and error message?